bats/tests/helpers/defaults.bash

```diff
@@ -1,14 +1,42 @@
+validate_enum() {
+    local var=$1
```

Comment:
```suggestion
    local var="$1"
```

Reply:
Why? It only makes a difference in places where the shell would otherwise perform word-splitting, but that is not the case for variable assignments:

```console
$ bar=$(echo one two)
$ foo=$bar printenv foo
one two
$ foo() { local var=$1; echo "$var"; }
$ foo one two
one
$ foo "one two"
one two
```

Now, we can make the argument that we should always quote shell variables, just so the reader doesn't have to know/understand when the shell will split words, and when it doesn't.

I'm fine with that, except that we just argued in the last retro that we don't want to call out style issues in code review for code that passes the linter and auto-formatter, and in this case both `shellcheck` and `shfmt` are fine with not quoting the variable.

I'm going to add the quotes, but leave it to you to figure out how to teach this rule to our tools, so that we can apply this consistently. Because a simplistic search finds at least another 19 lines (in addition to the 5 from this PR) where we don't quote variable assignments:

```
$ grep -r '=\$' tests | grep -v '"' | wc -l
      24
```

Reply:
I've made the changes, also to all other locations that didn't follow the "always quote substitutions" rule: https://github.com/rancher-sandbox/rancher-desktop/commit/40c2deac8053f7c1e9d21cb7aa14dcbcb1828dd1

Not really sure if this is a net gain when I look at things like this:

```diff
-PATH_BATS_HELPERS=$(absolute_path "$(dirname "${BASH_SOURCE[0]}")")
-PATH_BATS_ROOT=$(absolute_path "$PATH_BATS_HELPERS/../..")
+PATH_BATS_HELPERS="$(absolute_path "$(dirname "${BASH_SOURCE[0]}")")"
+PATH_BATS_ROOT="$(absolute_path "$PATH_BATS_HELPERS/../..")"
```

Please don't merge as-is; I have used separate commits for further review, but want to squash before merging.